server/consumer.go

```diff
@@ -2177,9 +2184,22 @@ func (mset *Stream) deliveryFormsCycle(deliverySubject string) bool {
 	return false
 }
 
-// This is same as check for delivery cycle.
+// Check that the subject is a subset of the stream's configured subjects,
+// or returns true if the stream has been created with no subject.
 func (mset *Stream) validSubject(partitionSubject string) bool {
-	return mset.deliveryFormsCycle(partitionSubject)
```

Comment:
This used anymore? If so move to `RLock` as well.

Reply:
Not sure what you mean, but yes, deliveryFormsCycle is still used in this branch. 
The thing is that I want validSubject() to return true if the stream is created with "no subject allowed", but if I were to put the check in deliveryFormsCycle I would have to return "true" (that it forms cycle) which does not make sense to me. This is why I originally I duplicated the code from deliveryFormsCycle but made it about validating the subject.

Reply:
Understood, just saying the lock should be an RLock in `deliveryFormsCycle` too. That was my mistake.